lib/trino-filesystem/src/test/java/io/trino/filesystem/TestLocation.java

```diff
@@ -235,7 +245,7 @@ private static void assertFileName(String locationString, String fileName)
     void testParentDirectory()
     {
         assertParentDirectory("scheme://userInfo@host/path/name", Location.of("scheme://userInfo@host/path"));
-        assertParentDirectory("scheme://userInfo@host:1234/name", Location.of("scheme://userInfo@host:1234"));
+        assertParentDirectory("scheme://userInfo@host:1234/name", Location.of("scheme://userInfo@host:1234/"));
```

Comment:
Can you add an assertion for getting the parent of the file system root
`scheme://userInfo@host:1234/`